### DISCUSSION TITLE
Upgrade Kong version 2.7.x directly to 3.x

### DISCUSSION THREAD

**Discussion by leehangwee** [Kong/kong#12931] [Help] [answered]
Would it be possible to upgrade Kong in DB-mode directly from 2.7.x to 3.x, without upgrading it to 2.8.2 first? Because the only difference in the migrations files between 2.8.x and 2.7.x is this [script](https://github.com/Kong/kong/blob/2.8.2/kong/db/migrations/core/015_270_to_280.lua) for the  creation of vaults_beta table to provide secrets management but we are not using that feature at all.

**Comment by chronolaw** [comment]
I think it is possible, but we must test it before we deploy it in the production environment.

**Comment by bungle** [comment, marked as answer]
All migrations should be carried over to 3.x, so it should be possible. Just that we don't actively support or test direct jumps, but there should be no reason why it should not work.

**Comment by jeremyjpj0916** [comment]
As an OSS user I was able to go from 2.8.x to 3.7.x with no issues in traditional mode.
